psycopg/psycopg/__init__.py

```diff
@@ -26,6 +26,13 @@
 from .client_cursor import AsyncClientCursor, ClientCursor
 from .connection_async import AsyncConnection
 
+try:
+    from .connection_async import AnyIOConnection  # noqa: F401
+except ImportError:
+    _anyio = False
+else:
+    _anyio = True
+
```

Comment:
Or even making `AnyIOConnection` a class failing with ImportError on initialization, and giving it a public, documented classmethod `AnyIOConnection.is_supported()` returning True of False.

Reply:
The issue with `AnyIOConnection: Optional[Type[AsyncConnection]]` is that this will leads to typing error in user code such as:
```
Item "None" of "Optional[Type[AsyncConnection[Any]]]" has no attribute "connect"
```

I'm not sure about the `is_supported()` idea. On the one hand, it feels right for the pipeline mode where "supported" means having installed the correct libpq version (not a Python package), on the other hand, for a plain Python dependency it seems a bit strange... I'll take a stab on the `ImportError` on initialization though.

Reply:
Probably you are right: we could assume the library always supported because we assume that the dependencies were installed correctly. We can also always create a class, but making it fail __init__ with ImportError if anyio is missing, which would make typing easier.

What do you think about it?

Reply:
Yes, that's what I'll try to do.